extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -178,6 +223,46 @@ export class AmpAutocomplete extends AMP.BaseElement {
     this.action_ = Services.actionServiceForDoc(this.element);
     this.viewport_ = Services.viewportForDoc(this.element);
 
+    const inputElements = childElementsByTag(this.element, 'input');
+    const textareaElements = childElementsByTag(this.element, 'textarea');
+    userAssert(
+      inputElements.length + textareaElements.length === 1,
+      `${TAG} should contain exactly one <input> or <textarea> child`
+    );
+    this.inputElement_ =
+      /** @type {!HTMLInputElement} */ (inputElements[0] ||
+      textareaElements[0]);
+    if (this.inputElement_.hasAttribute('type')) {
+      const inputType = this.inputElement_.getAttribute('type');
+      userAssert(
+        inputType === 'text' || inputType === 'search',
+        `${TAG} requires the "type" attribute to be "text" or "search" if present on <input>`
+      );
+    }
+
+    this.inline_ = this.element.hasAttribute('inline');
+    if (this.inline_) {
+      userAssert(
+        isExperimentOn(this.win, 'amp-autocomplete'),
+        `Experiment ${TAG} is not turned on for "inline" attr.`
+      );
+      this.trigger_ = this.element.getAttribute('inline');
+    }
+
+    if (this.element.hasAttribute('query')) {
```

Comment:
Ditto, I think you can do the same here.

Also associate `this.element` on the assertions (this makes the error display on the element with `#development=1`.

```js
userAssert(
  !this.query_ || isExperimentOn(this.win, 'amp-autocomplete'),
  `Experiment ${TAG} is not turned on for "query" attr. %s`,
  this.element
);
```